Modules/unicodedata.c

```diff
@@ -795,14 +797,83 @@ is_normalized(PyObject *self, PyObject *input, int nfc, int k)
         unsigned char quickcheck = record->normalization_quick_check;
 
         if (quickcheck & quickcheck_mask)
-            return 0; /* this string might need normalization */
+            return MAYBE; /* this string might need normalization */
         if (combining && prev_combining > combining)
-            return 0; /* non-canonical sort order, not normalized */
+            return NO; /* non-canonical sort order, not normalized */
         prev_combining = combining;
     }
-    return 1; /* certainly normalized */
+    return YES; /* certainly normalized */
 }
 
+/*[clinic input]
+unicodedata.UCD.is_normalized
+
+    self: self
+    form: str
```

Comment:
Would it be possible to get form as a Unicode object and then use _PyUnicode_EqualToASCIIId() API for best performances?

You have to declare identifiers using:

```
_Py_IDENTIFIER(NFC);
...
```

And then replace strcmp(form, "NFC")==0 with _PyUnicode_EqualToASCIIId(form, &PyId_NFC).

It avoids to encode form to UTF-8.